buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -229,9 +231,11 @@ class PrecommitTasks {
     }
 
     private static Task configureLoggerUsage(Project project) {
+        Object dependency = ClasspathUtils.isElasticsearchProject() ? project.project(':test:logger-usage') :
```

Comment:
I think having these conditionals everywhere will not only get frustrating, but also make mistakes. The nice thing about the dependency substitutions was we had one place they were defined.

I wonder if we could keep a static map of these project to dep, and move this conditional into there, and in places like this use a static util method with the project name?

Reply:
Unfortunately, it _can't_ be a static map because we have to return `Project` instances from this in the case of project dependencies, and as such, we need to hold on to an instance of `Project` which is a no-no. I think this is another thing we could refactor when we institute our "buildInfo" solution. As we would have to "initialize" this with the correct dependency types.

Reply:
I think it can still be static, as we only need the map to contain the project path, no the project object itself. Then the helper method would take in project and be able to look up the project by path. This could be a followup, it's not critical to this PR.

Reply:
> Then the helper method would take in project and be able to look up the project by path.

Right, we'd still have to pass a reference to the project. That's fine.

> This could be a followup, it's not critical to this PR.

We should do this as part of the `BuildPlugin` cleanup. Essentially `PrecommitTasks` should become it's own plugin.